### ISSUE TITLE
Unable to determine Node version

### ISSUE THREAD

**Issue by bennygenel** [onmodulus/modulus-cli#61]
While deploying meteor app modulus deploy gives error Unable to determine Node version and fail deploy. 

Everything was OK couple of days ago when I deployed same app with same code.

OS : Windows 8.1
Meteor version : 1.2.0.4
Node version : 5.0.0

```
Compressing project...
23.3 MB written
Uploading project...
Upload progress [===================] 100%
Deploying project into Node.js runtime...
Starting build.
Creating directories for build environment.
Downloading source.
Executing build.
Package found: /E/Website/Meteor/meteor 5/webstorm_cik/.demeteorized/bundle/programs/server/package.json
Unable to determine Node version. Verify package.json has no errors.
Build attempt failed, attempting again.
```

**Comment by fiveisprime** [comment]
What is the *.onmodulus.net URL for your project? I need to investigate this on the Modulus side.

**Comment by bennygenel** [comment]
Thanks for the reply. 
URL : democumle-53398.onmodulus.net

I created a new project to find out if it is a project only configuration problem or something but I got the same error on the new project too.

**Comment by bergerd1** [comment]
Hey Benny,

I tried deploying a copy of your project from the directory 'E' that your project appears to be running from.  I got the same error you had when attempting to deploy.

```
[~/E] # modulus deploy -p democumle-copy
Welcome to Modulus
[Warning] Your version 3.14.1 is behind the latest release 5.0.1.
Please update using "npm update -g modulus"
You are logged in as david.berger
[?] Project democumle-copy not found, do you want to create it? (no) yes
Creating project democumle-copy
Please choose which runtime to use:
  1) Node.js
  2) Java
  3) PHP
  4) Static
  5) Python (beta)
[?] Image number? 1
[?] Enter a servo size in MB [192, 396, 512, 1024, or 2048] (optional, default 512): 192
[✓] New project democumle-copy created.
Compressing project...
2.6 MB written
Uploading project...
Upload progress [===================] 100%
Deploying project into Node.js runtime...
Starting build.
Creating directories for build environment.
Downloading source.
Executing build.
Package found: /Website/Meteor/meteor 5/webstorm_cik/.demeteorized/bundle/programs/server/package.json
Unable to determine Node version. Verify package.json has no errors.
Build attempt failed, attempting again.
Starting build.
Creating directories for build environment.
Upload progress [===================] 100%Downloading source.
Executing build.
Package found: /Website/Meteor/meteor 5/webstorm_cik/.demeteorized/bundle/programs/server/package.json
Unable to determine Node version. Verify package.json has no errors.
Build attempt failed, attempting again.
Starting build.
Creating directories for build environment.
Downloading source.
Executing build.
Package found: /Website/Meteor/meteor 5/webstorm_cik/.demeteorized/bundle/programs/server/package.json
Unable to determine Node version. Verify package.json has no errors.
Build failed, reverting project.

[Error] There was a problem deploying your project.

[~/E] #
```

However, when I changed directory to the root directory of the project it deployed fine.  

```
[~/E/website/meteor/meteor 5/webstorm_cik/.demeteorized/bundle] # modulus deploy -p democumle-copy
Welcome to Modulus
[Warning] Your version 3.14.1 is behind the latest release 5.0.1.
Please update using "npm update -g modulus"
You are logged in as david.berger
Selecting democumle-copy

Compressing project...
2.6 MB written
Uploading project...
Upload progress [===================] 100%
Deploying project into Node.js runtime...
Starting build.
Creating directories for build environment.
Downloading source.
Executing build.
Package found: /programs/server/package.json
Installing node 0.10.40
Installing npm 3.3.10
Installing packages from /programs/server/package.json
WARN: npm-shrinkwrap.json will override dependencies declared in package.json.

> fibers@1.0.5 install /mnt/input/programs/server/node_modules/fibers
> node ./build.js

make: Entering directory `/mnt/input/programs/server/node_modules/fibers/build'
  CXX(target) Release/obj.target/fibers/src/fibers.o
  CXX(target) Release/obj.target/fibers/src/coroutine.o
../src/coroutine.cc: In function ‘void* find_thread_id_key(void*)’:
../src/coroutine.cc:64:53: warning: comparison of unsigned expression >= 0 is always true [-Wtype-limits]
  for (pthread_key_t ii = coro_thread_key - 1; ii >= 0; --ii) {
                                                     ^
../src/coroutine.cc:90:3: warning: ‘thread_id’ may be used uninitialized in this function [-Wmaybe-uninitialized]
   if (tls == thread_id) {
   ^
  CC(target) Release/obj.target/fibers/src/libcoro/coro.o
  SOLINK_MODULE(target) Release/obj.target/fibers.node
  COPY Release/fibers.node
make: Leaving directory `/mnt/input/programs/server/node_modules/fibers/build'
Installed in `/mnt/input/programs/server/node_modules/fibers/bin/linux-x64-v8-3.14/fibers.node`
meteor-dev-bundle@0.0.0 /mnt/input/programs/server
├── ansi-regex@0.2.1
├── ansi-styles@1.1.0
├── escape-string-regexp@1.0.3
├── fibers@1.0.5
├── has-ansi@0.1.0
├─┬ meteor-promise@0.5.0
│ └─┬ promise@7.0.4
│   └── asap@2.0.3
├── semver@4.1.0
├─┬ source-map-support@0.3.2
│ └─┬ source-map@0.1.32
│   └── amdefine@1.0.0
├── strip-ansi@0.3.0
├── supports-color@0.2.0
├── type-of@2.0.1
└── underscore@1.5.2

npm WARN EPACKAGEJSON meteor-dev-bundle@0.0.0 No description
npm WARN EPACKAGEJSON meteor-dev-bundle@0.0.0 No repository field.
npm WARN EPACKAGEJSON meteor-dev-bundle@0.0.0 No license field.
v0.10.40
deploy -> v0.10.40
Uploading completed build.
Provisioning and deploying bundle to servos...

[✓] democumle-copy running at democumlecopy-54507.onmodulus.net

[~/E/website/meteor/meteor 5/webstorm_cik/.demeteorized/bundle] #
```

**Comment by bennygenel** [comment]
Thanks for your reply.

After I received your message I tried to deploy again. I deploy from the root of the project folder as always but seemed like something is wrong with my computer. So I created another meteor project. I watched a video on Meteor channel a couple of days ago. It was about the Meteor Windows Installer and they were talking about the long file names etc. I thought maybe this is happening because of that.
Anyway after I copied the old project files to new project and tried to deploy it deployed without any errors. I guess I was right or just got lucky.

Thanks for your time

Take care
